dashboard/app/models/queued_account_purge.rb

```diff
@@ -40,7 +43,13 @@ class QueuedAccountPurge < ApplicationRecord
   # Used by developers to resolve an account purge queued for manual review,
   # after they've investigated the account and decided it's ready to purge.
   def resolve!
-    AccountPurger.new(bypass_safety_constraints: true).purge_data_for_account User.with_deleted.find(user_id)
+    AccountPurger.new(bypass_safety_constraints: true).purge_data_for_account user
     destroy!
   end
+
+  # It's possible to have a QueuedAccountPurge still around, pointing at an account that
+  # has already been purged.  This method finds and removes those records.
+  def self.clean_up_resolved_records!
+    joins(:user).where.not(users: {purged_at: nil}).destroy_all
```

Comment:
```suggestion
    # Not too worried about efficiency here because:
    # a) It runs once every night, close to nadir traffic.
    # b) Number of selected records should always be double-digits and below.
    joins(:user).where.not(users: {purged_at: nil}).destroy_all
```